src/metal.cs

```diff
@@ -906,13 +1111,61 @@ public partial interface MTLVertexAttribute {
 
 		[Export ("name")]
 		string Name { get; }
+		
+		[iOS (10, 0), TV (10,0), NoWatch, Mac (10,12)]
```

Comment:
mmm, I think you are missing `onlyOn64 : true` on all your `Mac (10,12)` attributes -> `Mac (10,12, onlyOn64 : true)]` since Metal is a 64 bits only API IIRC